sqlmesh/dbt/model.py

```diff
@@ -115,22 +116,10 @@ def _validate_partition_by(cls, v: t.Any) -> t.Union[t.List[str], t.Dict[str, t.
         **BaseModelConfig._FIELD_UPDATE_STRATEGY,
         **{
             "sql": UpdateStrategy.IMMUTABLE,
-            "time_column_": UpdateStrategy.IMMUTABLE,
+            "time_column": UpdateStrategy.IMMUTABLE,
         },
     }
 
-    @property
-    def time_column(self) -> t.Optional[str]:
```

Comment:
Is the rationale for removing this that if someone is switching to `sqlmesh_incremental`, we'll want them to add the `time_column` param too?

Reply:
Yes, that would make their intent explicit

Reply:
Please update the dbt integration docs to reflect this. It's a bit dangerous because the user will not get an error if they have a mismatch (time_column, but not {% sqlmesh_incremental is defined %}. If they get it wrong, they are basically doing a FULL kind.

Reply:
Going to update the documentation once all relevant changes are in.